wled00/FX_fcn.cpp

```diff
@@ -438,18 +438,21 @@ uint32_t IRAM_ATTR Segment::currentColor(uint8_t slot) {
 #endif
 }
 
-CRGBPalette16 IRAM_ATTR &Segment::currentPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
-  loadPalette(targetPalette, pal);
-  unsigned prog = progress();
+inline CRGBPalette16 &Segment::currentPalette(uint8_t pal) {
+  if ( SEGPALETTEIDX != pal ) {
+    loadPalette(SEGPALETTE, pal);
+    SEGPALETTEIDX = pal;
+  }
+  uint16_t prog = progress();
   if (strip.paletteFade && prog < 0xFFFFU) {
     // blend palettes
     // there are about 255 blend passes of 48 "blends" to completely blend two palettes (in _dur time)
     // minimum blend time is 100ms maximum is 65535ms
-    unsigned noOfBlends = ((255U * prog) / 0xFFFFU) - _t->_prevPaletteBlends;
-    for (unsigned i=0; i<noOfBlends; i++, _t->_prevPaletteBlends++) nblendPaletteTowardPalette(_t->_palT, targetPalette, 48);
-    targetPalette = _t->_palT; // copy transitioning/temporary palette
+    uint16_t noOfBlends = ((255U * prog) / 0xFFFFU) - _t->_prevPaletteBlends;
```

Comment:
`uint16_t` will unnecessarily increase code size and slow execution down as it adds `& 0xFFFF` to every operation.

Reply:
Good catch, we can keep both vars as 'unsigned'